adafruit_gps.py

```diff
@@ -34,6 +34,140 @@
 
 _GPSI2C_DEFAULT_ADDRESS = const(0x10)
 
+_SENTENCE_PARAMS = {
```

Comment:
Any idea how many bytes this grows the library? `gc.mem_free()` returns the number of free bytes. I'm worried this will fill up the SAMD21's RAM.

Reply:
[memory.txt](https://github.com/adafruit/Adafruit_CircuitPython_GPS/files/6046322/memory.txt)
The attached file lists memory Avg/Min/Max.
It averages ~78500 free after 500 cycles. (PyPortal Pynt SAMD51)
It's running from a .py file, not a .mpy file if that makes a difference.

The _SENTENCE_PARAMS structure could be replaced with a byte array if that will reduce size. I'm not used to considering memory usage.

Reply:
Sorry, I should have been clearer. Please measure the memory free before the import and after and compare it to the memory free values before these changes. mpy may be smaller so I'd suggest using that. I'm worried this will add 1k+ that will make it harder to import on SAMD21s.

Reply:
I'm much less worried about memory usage over time because the objects used tend to be freed quickly. Everything loaded on import lasts until the end of the program.

Reply:
```
from gc import mem_free
mem01 = mem_free()
print(f"{mem01} bytes free after mem_free import")

import adafruit_gps
mem02 = mem_free()
gps_version = adafruit_gps.__version__
print(f"{mem02} bytes free after adafruit_gps {gps_version} import")

diff = mem01 - mem02
print(f"Difference: {diff}")
```
.mpy
```
219440 bytes free after mem_free import
211840 bytes free after adafruit_gps 3.7.0 import
Difference: 7600
```

.py
```
219440 bytes free after mem_free import
209744 bytes free after adafruit_gps 0.0.0-auto.0 import
Difference: 9696
```

.mpy
```
219296 bytes free after mem_free import
210128 bytes free after adafruit_gps 0.0.0-auto.0 import
Difference: 9168
```
That's a 1568 byte increase in the .mpy. How small a target do you suggest?

Reply:
```
219168 bytes free after mem_free import
210800 bytes free after adafruit_gps 0.0.0-auto.0 import
Difference: 8368
```
I've got it down to the above, an increase of 768 bytes from the last release. I've not pushed the commit to github yet as the CI will fail.

Reply:
Under 1k seems reasonable to me. The main thing is to be aware of it. Thanks for the testing and improvements!